Config/SEGGER_RTT_Conf.h

```diff
@@ -152,10 +152,10 @@ Revision: $Rev: 24316 $
 #if ((defined(__SES_ARM) || defined(__SES_RISCV) || defined(__CROSSWORKS_ARM) || defined(__GNUC__) || defined(__clang__)) && !defined (__CC_ARM) && !defined(WIN32))
   #if defined(__ZEPHYR__) && defined (CONFIG_SEGGER_RTT_CUSTOM_LOCKING)
     #ifdef CONFIG_MULTITHREADING
```

Comment:
If you make this change you don't need this `#ifdef` anymore. You can just use `k_spin_(un)lock()` regardless of multithreading or not.

Reply:
Note: `k_spin_(un)lock` maps directly to `irq_(un)lock` in uniprocessor systems.